src/compiler/compile/render_dom/index.ts

```diff
@@ -425,11 +425,15 @@ export default function dom(
 		`);
 	}
 
-	const prop_names = x`[]`;
+	const prop_names = x`{}` as ObjectExpression;
```

Comment:
Untested, but this should be able to be tidied up a bit, to something like

```js
const prop_names = x`{ ${props.map(v => p`${v.export_name}: ${v.export_name === v.name ? x`0` : x`"${v.name}"`}`) }`;
```

This seems reasonable though. I think the slight weirdness is worth it.

Reply:
good idea — updated